add.go

```diff
@@ -188,10 +188,39 @@ func includeDirectoryAnyway(path string, pm *fileutils.PatternMatcher) bool {
 	return false
 }
 
+// validateSourceAndDest performs a sanity check on the case that if more
+// than one source is provided then destination must end with a slash (`/`),
+// which signifies that destination is a directory and similarly function
+// returns error if destination does not ends with a slash (`/`).
+// Following function can contain more validation cases in future.
+func validateSourceAndDest(destination string, sources []string) error {
+	multipleSources := false
+	if len(sources) > 1 {
+		multipleSources = true
+	} else {
+		// Check length before accessing `0`
+		// to avoid `index out of range`.
+		if len(sources) == 1 {
+			if strings.HasSuffix(sources[0], "*") {
```

Comment:
Should you check for len(sources)==0 and return an error?

Reply:
Why is a suffix of "*" the special case here?

Reply:
Special case is here to match with the current behavior of docker https://github.com/containers/buildah/issues/4167#issuecomment-1212098545